forms/error-helpers.js

```diff
@@ -21,6 +21,48 @@ define(function (require) {
     };
   };
 
+  function makeCustomError(code, val, errorString){
+    var ret = {};
+
+    if ( errorString === undefined){
+      errorString = blinkFormsError.toErrorString(val) || val;
+    }
+
+    ret.code = code;
+    ret[code] = val;
```

Comment:
We have examples of translation strings where the variable name doesn't match the error code:
- https://github.com/blinkmobile/forms/blob/develop/forms/locales/en/BMP/Forms/validation.json
Arguably, we should fix this there instead of in this custom function, but it's something to be aware of.
Also, whilst we haven't done so yet, it's foreseeable that a translation string might have multiple variables, at least one of which won't be matching the error code.

This is a private helper function, so it's probably fine to leave it the way it is. I just wanted to note my thoughts on this somewhere so we can find it later.